src/server/pkg/storage/fileset/fileset.go

```diff
@@ -52,7 +53,12 @@ func newFileSet(ctx context.Context, storage *Storage, name string, memThreshold
 }
 
 // Put reads files from a tar stream and adds them to the fileset.
-func (f *FileSet) Put(r io.Reader) error {
+// (bryce) probably should prevent / clean files that end with "/", since that will indicate a directory.
+func (f *FileSet) Put(r io.Reader, customTag ...string) error {
+	tag := f.defaultTag
+	if len(customTag) > 0 {
+		tag = customTag[0]
+	}
```

Comment:
similarly here, consider erroring if more than one `customTag` is passed in

Reply:
Yeah, none of the optional parameter code I have seen checks if it is too long, and none of mine does. I'm not too sure if it is really worth the extra code.